src/plugins/initiatives/parseInitiative.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+import * as C from "../../util/combo";
+import {compatibleParser} from "../../util/compat";
+import {COMPAT_INFO, upgradeFrom010} from "./initiativeFile";
+import type {
+  InitiativeFile,
+  InitiativeFileV010,
+  InitiativeFileV020,
+} from "./initiativeFile";
+import {_validateUrl} from "./initiativesDirectory";
+import {fromISO, toISO} from "../../util/timestamp";
+
+const URLParser = C.fmap(C.string, _validateUrl);
+
+const TimestampParser = C.fmap(C.string, (t) => toISO(fromISO(t)));
+
+const CommonFields = {
+  title: C.string,
+  timestampIso: TimestampParser,
+  weight: C.object({incomplete: C.number, complete: C.number}),
+  completed: C.boolean,
+};
+
+const Parse_020: C.Parser<InitiativeFileV020> = (() => {
+  const NodeEntryParser = C.object(
+    {
+      title: C.string,
+      timestampIso: TimestampParser,
+      contributors: C.array(URLParser),
+    },
+    {
+      key: C.string,
+      weight: C.number,
+    }
+  );
+
+  const EdgeSpecParser = C.object({
+    urls: C.array(URLParser),
+    entries: C.array(NodeEntryParser),
+  });
+
+  return C.object(CommonFields, {
+    contributions: EdgeSpecParser,
+    dependencies: EdgeSpecParser,
+    references: EdgeSpecParser,
+    champions: C.array(URLParser),
+  });
+})();
```

Comment:
I generally prefer to assemble "bare" parsers (not inside an IIFE) and compose them outside of a function scope. Think its a bit simpler. Fine with this too, though (no change requested).

Reply:
yea, I just noticed a bunch of the other parsers were defined as IIFEs and copied that style since I thought that was preferred for some reason lol, was also confused.